tests/kokkos-based/symmetric_matrix_rank1_update_kokkos.cpp

```diff
@@ -0,0 +1,305 @@
+    const mdspan<ElementType2, extents<dynamic_extent>, LayoutPolicy2, AccessorPolicy2> &v2)
+{
+  const auto size = v1.extent(0);
+  if (size != v2.extent(0))
+    return false;
+  int diff = false;
+  Kokkos::parallel_reduce(size,
+    KOKKOS_LAMBDA(const std::size_t i, decltype(diff) &d){
+        d = d || !(v1(i) == v2(i));
+	    }, diff);
+  return !diff;
+}
+
+template <typename ElementType1,
+          typename LayoutPolicy,
+          typename AccessorPolicy,
+          typename ElementType2>
+inline bool is_same_vector(
+    const mdspan<ElementType1, extents<dynamic_extent>, LayoutPolicy, AccessorPolicy> &v1,
+    const std::vector<ElementType2> &v2)
+{
+  return is_same_vector(v1, make_mdspan(v2));
+}
+
+template <typename ElementType1,
+          typename LayoutPolicy,
+          typename AccessorPolicy,
+          typename ElementType2>
+inline bool is_same_vector(
+    const std::vector<ElementType1> &v1,
+    const mdspan<ElementType2, extents<dynamic_extent>, LayoutPolicy, AccessorPolicy> &v2)
+{
+  return is_same_vector(v2, v1);
+}
+
+template <typename ElementType>
+inline bool is_same_vector(
+    const std::vector<ElementType> &v1,
+    const std::vector<ElementType> &v2)
+{
+  return is_same_vector(make_mdspan(v1), make_mdspan(v2));
+}
+
+// real diff: d = |v1 - v2|
+template <typename T, typename enabled=void>
+class value_diff {
+public:
+  value_diff(const T &val1, const T &val2): _v(fabs(val1 - val2)) {}
+  operator T() const { return _v; }
+protected:
+  value_diff() = default;
+  T _v;
+};
+
+// real diff: d = max(|R(v1) - R(v2)|, |I(v1) - I(v2)|)
+// Note: returned value is of underlying real type
+template <typename T>
+class value_diff<std::complex<T>>: public value_diff<T> {
+  using base = value_diff<T>;
+public:
+  value_diff(const std::complex<T> &val1, const std::complex<T> &val2) {
+    const T dreal = base(val1.real(), val2.real());
+    const T dimag = base(val1.imag(), val2.imag());
+    base::_v = std::max(dreal, dimag);
+  }
+};
+
+template <typename ElementType,
+          typename LayoutPolicy1,
+          typename AccessorPolicy1,
+          typename LayoutPolicy2,
+          typename AccessorPolicy2,
+          typename ToleranceType>
+inline bool is_same_matrix(
+    const mdspan<ElementType, extents<dynamic_extent, dynamic_extent>, LayoutPolicy1, AccessorPolicy1> &A,
+    const mdspan<ElementType, extents<dynamic_extent, dynamic_extent>, LayoutPolicy2, AccessorPolicy2> &B,
+    ToleranceType tolerance)
+{
+  const auto ext0 = A.extent(0);
+  const auto ext1 = A.extent(1);
+  if (B.extent(0) != ext0 or B.extent(1) != ext1)
+    return false;
+  int diff = false;
+  const auto size = ext0 * ext1;
+  Kokkos::parallel_reduce(size,
+    KOKKOS_LAMBDA(const std::size_t ij, decltype(diff) &d) {
+        const auto i = ij / ext1;
+        const auto j = ij - i * ext1;
+        if (value_diff(A(i, j), B(i, j)) > tolerance)
```

Comment:
Does this actually work on the GPU?  It's using `std::complex`.

Reply:
I tried building for CUDA but failed with `nvcc` not being able to compile `mdspan` (the errors didn't look fatal, but I was not sure if it's worth digging into).
Would you know if `stdBLAS` should build for CUDA regardless of requiring C++20 while [`nvcc` only supports C++17](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#options-for-altering-compiler-linker-behavior-std) ?

Reply:
Oh, maybe it was about `nvcc` complaining `std::complex` - I'll check that.

Reply:
Probably it's also not a good idea to use parallel elements in tests -
or do it through `ParallelMatrixVisitor` with `is_int_mul_safe` (see the discussion [above](https://github.com/kokkos/stdBLAS/pull/167#discussion_r846204678))

Reply:
Fixed with https://github.com/kokkos/stdBLAS/pull/167/commits/83d174227fd8cd9aebf11c76e14646401f713156: we cast `mdspan`s to `Kokkos::View`s correctly now, so inside lambda we'll have `std::complex` replaced with `Kokkos::complex` by `mdspan_to_view()`.

@mhoemmen thank you for pointing this out!